utility/utils.py

```diff
@@ -750,7 +749,63 @@ def download_openshift_installer(version=defaults.INSTALLER_VERSION):
             f"https://mirror.openshift.com/pub/openshift-v4/clients/ocp/"
             f"{version}/openshift-install-{os_type}-{version}.tar.gz"
         )
+        # Prepare BIN_DIR
+        try:
+            os.mkdir(defaults.BIN_DIR)
+        except FileExistsError:
+            pass
+        # record current working directory and switch to BIN_DIR
+        previous_dir = os.getcwd()
+        os.chdir(defaults.BIN_DIR)
+        tarball = f"{installer_filename}.tar.gz"
+        download_file(url, tarball)
+        run_cmd(f"tar xzvf {tarball}")
+        os.remove(tarball)
+        # return to the previous working directory
+        os.chdir(previous_dir)
+
+    return installer_binary_path
+
+def get_openshift_client(version=defaults.CLIENT_VERSION):
+    """
+    Get path of the openshift client binary, download it if not available.
+
+    Args:
+        version (str): version of the client to download
+
+    Returns:
+        str: path of the client binary
+
+    """
+    client_binary_path = os.path.join(defaults.BIN_DIR, 'oc')
+    if os.path.isfile(client_binary_path):
+        log.info("Client exists, skipping download")
+        # TODO: check client version
+    else:
+        log.info("Downloading openshift client")
+        if platform.system() == "Darwin":
+            os_type = "mac"
+        elif platform.system() == "Linux":
+            os_type = "linux"
+        else:
+            raise UnsupportedOSType
+        url = (
+            f"https://mirror.openshift.com/pub/openshift-v4/clients/ocp/"
```

Comment:
I see taht path is almost same as here: https://github.com/red-hat-storage/ocs-ci/pull/49/files#diff-ef93c603b643a8362ba61f8d9797bac2R749, so I would move this URL to some different place with format varaibles and would format it on this place and for installer as well with provided variables.

Reply:
Updated in 0056ce6f3e59d0365c7ef238bd4b29cd3b1e36bc.